### PR TITLE
Add protocol v4.

### PR DESCRIPTION
**Pull request by exazg**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,19 +15,19 @@ ODBC or ADO.NET, which are mostly old and static standards and create
 additional complexity due to the necessary driver managers.
 
 ## Changes
+* [WebSocket API v4](docs/WebsocketAPIV4.md) support has been added in Exasol 8.31.0.
 * Autocommit was enabled by default for sessions in Exasol 7.1+.
 * [WebSocket API v3](docs/WebsocketAPIV3.md) support has been added in Exasol 7.1.
-* [WebSocket API v2](docs/WebsocketAPIV2.md) support has been added in Exasol 7.0.
```
Comment (exa-mohamedi) [inline README.md:20]: Is it intended to be removed?

File: docs/WebsocketAPIV4.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+### Connection-related commands
+
+The following commands are used to connect to Exasol, disconnect from Exasol, 
+and query the hosts of an Exasol cluster.
+
+| Command | Description |
+| --- | --- |
+| [disconnect](commands/disconnectV1.md) | Closes a connection to Exasol |
+| [enterParallel](commands/enterParallelV1.md) | Opens subconnections for parallel execution<br>:warning: Deprecated, please use [requestParallelConnections](commands/requestParallelConnectionsV4.md) instead. |
```
Comment (exa-mohamedi) [inline docs/WebsocketAPIV4.md:40]: Can we remove it here?

File: README.md (hunk 1 above)
Comment (exazg) [reply README.md:20]: I removed it because 7.0 isn't supported anymore.

File: docs/WebsocketAPIV4.md (hunk 2 above)
Comment (exazg) [reply docs/WebsocketAPIV4.md:40]: It is still usable, isn't it? Or was it removed from v4?

Comment (exazg) [reply docs/WebsocketAPIV4.md:40]: It looks like it's still available in v4.

File: docs/commands/subLoginTokenV4.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    }
+   ```
+   
+
+3. The client sends either an OpenID `accessToken` or a `refreshToken` and optionally other client information.
+
+   Request fields:
+     * accessToken (string, optional) => OpenID access token to use for the login process
+     * refreshToken (string, optional) => OpenID refresh token to use for the login process
+     * token (number) => token required for subconnection logins (see, EnterParallel)
```
Comment (exa-mohamedi) [inline docs/commands/subLoginTokenV4.md:55]: Can we update the EnterParallel to requestParallelConnectionsV4?

Comment (exazg) [reply docs/commands/subLoginTokenV4.md]: Ah, good find. Fixed.

File: docs/WebsocketAPIV4.md (hunk 2 above)
Comment (exa-mohamedi) [reply docs/WebsocketAPIV4.md:40]: Yes. It can be used in v4. I taught as it is deprecated we can remove it from v4.

File: README.md (hunk 1 above)
Comment (exa-mohamedi) [reply README.md:20]: ok.

File: docs/WebsocketAPIV4.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+| [getHosts](commands/getHostsV1.md) | Gets the hosts in a cluster |
+| [login](commands/loginV3.md) | Establishes a connection to Exasol |
+| [loginToken](commands/loginTokenV3.md) | Establishes a connection to Exasol using an OpenID token |
+| [requestParallelConnections](commands/requestParallelConnectionsV4.md) | Request subconnections for parallel execution |
+| [subLogin](commands/subLoginV3.md) | Establishes a subconnection to Exasol |
```
Comment (exa-mohamedi) [inline docs/WebsocketAPIV4.md:45]: Can we redirect to commans/subLoginV4.md?

File: docs/WebsocketAPIV4.md [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+| [subLoginToken](commands/subLoginTokenV3.md) | Establishes a subconnection to Exasol using an OpenID token |
```
Comment (exa-mohamedi) [inline docs/WebsocketAPIV4.md:46]: Similarly, Can we redirect to commans/subLoginTokenV4.md?

File: README.md [hunk 6]
```diff
@@ -15,19 +15,19 @@ ODBC or ADO.NET, which are mostly old and static standards and create
 additional complexity due to the necessary driver managers.
 
 ## Changes
+* [WebSocket API v4](docs/WebsocketAPIV4.md) support has been added in Exasol 8.31.0.
```
Comment (exa-mohamedi) [inline README.md:18]: If DB Exa to Exa changes are merged this week, then DB version will be 8.32.0.

File: docs/WebsocketAPIV4.md [hunk 7]
```diff
@@ -0,0 +1,235 @@
+## WebSocket protocol v4 details
+
+WebSocket Protocol v4 requires an Exasol version of at least 8.31.0. It follows the RFC 6455 document of the IETF.
```
Comment (exa-mohamedi) [inline docs/WebsocketAPIV4.md:3]: If DB Exa to Exa changes are merged this week, then DB version will be 8.32.0.

File: docs/WebsocketAPIV4.md [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+* [Data Types: Type names and properties](#data-types-type-names-and-properties)
+* [Compression](#compression)
+* [Heartbeat/Feedback messages](#heartbeatfeedback-messages)
+* [Subconnections](#subconnections)
+
+## Changes
+
+| Date | Exasol Version | Change | Issue |
+| --- | --- | --- | --- |
+| 2024.09.05 | 8.31.0 | The latest update to Exasol enables the use of the database port for subconnections instead of requiring ports in the range 20000 to 21000. Additionally, this update eliminates the need for special IP address configurations on the database. | |
```
Comment (exa-mohamedi) [inline docs/WebsocketAPIV4.md:27]: If DB Exa to Exa changes are merged this week, then DB version will be 8.32.0.

File: docs/WebsocketAPIV4.md [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+| 2024.09.05 | 8.31.0 | The `enterParallel` command has been deprecated. | |
```
Comment (exa-mohamedi) [inline docs/WebsocketAPIV4.md:28]: If DB Exa to Exa changes are merged this week, then DB version will be 8.32.0.

File: docs/WebsocketAPIV4.md (hunk 4 above)
Comment (exazg) [reply docs/WebsocketAPIV4.md]: Done

File: README.md (hunk 6 above)
Comment (exazg) [reply README.md]: Done

File: docs/WebsocketAPIV4.md (hunk 7 above)
Comment (exazg) [reply docs/WebsocketAPIV4.md]: Done

File: docs/WebsocketAPIV4.md (hunk 5 above)
Comment (exazg) [reply docs/WebsocketAPIV4.md]: Done

File: docs/WebsocketAPIV4.md (hunk 8 above)
Comment (exazg) [reply docs/WebsocketAPIV4.md]: Done

File: docs/WebsocketAPIV4.md (hunk 9 above)
Comment (exazg) [reply docs/WebsocketAPIV4.md]: Done

**Review by exa-mohamedi — APPROVED** [review]

**Pull request merged**